astropy/coordinates/representation.py

```diff
@@ -221,6 +226,138 @@ def _unitstr(self):
                            for component in self.components]))
         return unitstr
 
+    def _scale_operation(self, op, *args):
+        results = []
+        for component, cls in self.attr_classes.items():
+            value = getattr(self, component)
+            if issubclass(cls, Angle):
+                results.append(value)
+            else:
+                result = op(value, *args)
+                if result is NotImplemented:
+                    return NotImplemented
+                else:
+                    results.append(result)
+
+        return self.__class__(*results)
+
+    def __mul__(self, other):
+        return self._scale_operation(operator.mul, other)
+
+    def __rmul__(self, other):
+        return self.__mul__(other)
+
+    def __truediv__(self, other):
+        return self._scale_operation(operator.truediv, other)
+
+    def __div__(self, other):
+        return self._scale_operation(operator.truediv, other)
+
+    def __neg__(self):
+        return self._scale_operation(operator.neg)
+
+    # Follow numpy convention and make an independent copy.
+    def __pos__(self):
+        return self.__class__(*(getattr(self, component)
+                                for component in self.components), copy=True)
+
+    def __abs__(self):
```

Comment:
If we keep this, I'm also curious here why you use `functools` and not:
```python
return np.sqrt(np.sum([getattr(a, component)**2
                           for component in a.components], axis=0))
```
(which is a bit faster I think)